### PR TITLE
Enable PSA on 1.26 lanes

### PR DESCRIPTION
**Pull request by acardace**
Since https://github.com/kubevirt/kubevirt/pull/8649 KubeVirt can run VMs as `restricted` PSA pods, we should run all the latest lanes with PSA enabled to make sure we don't break this.

Running with PSA enabled translates in testing KubeVirt in the most restrictive Kubernetes environment and thus it's only more beneficial if we run every PR against PSA lanes.

Signed-off-by: Antonio Cardace <acardace@redhat.com>

### CODE REVIEW

**Comment by xpivarc** [comment]
@acardace Did we check these lanes pass?

**Comment by dhiller** [comment]
Regarding ipv6 failure: the lane seems to be broken, since a test run on it showed the same symptoms: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/9066/pull-kubevirt-e2e-k8s-1.26-ipv6-sig-network/1615638876666204160

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml [hunk 1]
```diff
@@ -491,7 +491,7 @@ presubmits:
         - /usr/local/bin/runner.sh
         - /bin/sh
         - -c
-        - cd cluster-provision/k8s/1.26 && ../provision.sh
+        - cd cluster-provision/k8s/1.26 && KUBEVIRT_PSA='true' ../provision.sh
```
Comment (dhiller) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml:494]: Sorry for this late remark, but I think you should do this in a separate PR, since we can't change the kubevirt/kubevirt lanes to enable psa before the psa is working inside kubevirt.

Comment (acardace) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml]: Gotcha, removed it!

Comment (dhiller) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml]: Thanks! LMK when you have created the other PR.

@brianmcarey I think we might set the psa check-provisions to `always_run: true` so we get any failures. WDYT?

Comment (brianmcarey) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml]: Yes I think we should have it set to always run. It is an important provider for kubevirt.

**Comment by dhiller** [comment]
https://github.com/kubevirt/kubevirtci/pull/950 has passed tests, now we wait for publish of kubevirtci to succeed, then the bump to happen and get merged.

**Comment by dhiller** [comment]
https://github.com/kubevirt/kubevirt/pull/9081 is waiting for merge.

**Comment by dhiller** [comment]
@acardace I wasn't expecting so many failures.

I think it's not the best idea to take this in right away, since it will create a lot of friction.

I'd suggest to duplicate the lane definitions and make them triggerable manually by setting `always_run: false`, `skip_report: true` (see docs [here](https://docs.prow.k8s.io/docs/jobs/)), so that other PR authors aren't blocked by these changes right away.

This would enable people to work on PSA, since they then can trigger the lanes by commenting with `/test pull-kubevirt-e2e-k8s-1.26-psa-sig-network` as usual.

**Comment by dhiller** [comment]
What do others think? @xpivarc @enp0s3 @brianmcarey

**Comment by acardace** [comment]
> @acardace I wasn't expecting so many failures.
me too

> 
> I think it's not the best idea to take this in right away, since it will create a lot of friction.
> 
> I'd suggest to duplicate the lane definitions and make them triggerable manually by setting `always_run: false`, `skip_report: true` (see docs [here](https://docs.prow.k8s.io/docs/jobs/)), so that other PR authors aren't blocked by these changes right away.
> 
> This would enable people to work on PSA, since they then can trigger the lanes by commenting with `/test pull-kubevirt-e2e-k8s-1.26-psa-sig-network` as usual.

What about enforcing the PSA compute lane and making the others optional as you described?

**Comment by xpivarc** [comment]
> > @acardace I wasn't expecting so many failures.
> 
> me too
> 
> > I think it's not the best idea to take this in right away, since it will create a lot of friction.
> > I'd suggest to duplicate the lane definitions and make them triggerable manually by setting `always_run: false`, `skip_report: true` (see docs [here](https://docs.prow.k8s.io/docs/jobs/)), so that other PR authors aren't blocked by these changes right away.
> > This would enable people to work on PSA, since they then can trigger the lanes by commenting with `/test pull-kubevirt-e2e-k8s-1.26-psa-sig-network` as usual.
> 
> What about enforcing the PSA compute lane and making the others optional as you described?

Sounds good.  Looking at the failures it should be pretty straightforward to fix them. The only problem will be the operator lane. There I think what is happening is the following: We downgrade Kubevirt but we have PSA enabled. The n-1 version is not PSA compliant and therefore it fails to run any VM.

**Review by dhiller — APPROVED** [review]
/approve

**Comment by acardace** [comment]
@dhiller can you remove the hold?
